### PR TITLE
pretty url

### PR DESCRIPTION
**Pull request by vickymesalcorta**
Dinu, el test del controller UserGiftRequestsController tenia escrito dos veces el mismo código. (Yo pensaba que lo habías borrado el otro dia) asi que borré la repetición de más abajo.
Una vez borrado eso y corridos los tests pinchan aqui:

Failures:

  1) UserGiftRequestsController#cancel When logged When user owns the gift item destroys the gift request
     Failure/Error: (expect do
       expected result to have changed by -1, but was changed by 0
     # ./spec/controllers/user_gift_requests_controller_spec.rb:28:in `block (5 levels) in <top (required)>'

  2) UserGiftRequestsController#cancel When logged When user owns the gift item sends an email
     Failure/Error: (expect do
       expected result to have changed by 1, but was changed by 0
     # ./spec/controllers/user_gift_requests_controller_spec.rb:34:in `block (5 levels) in <top (required)>'

la route para el metodo cancel es:
cancel_gift_request_organization_gift_requests DELETE     /users/organizations/:organization_id/gift_requests/:gift_item_id/cancel(.:format) user_gift_requests#cancel

Y el método del controller es:

  def cancel
    gift_request.destroy
    send_cancelation_email
    redirect_to :back
  end

en la primera linea se hace gift_request es decir se llama al método:
  private
  def gift_request
    @gift_request ||= GiftRequest.find_by_id(params[:id])
  end

No entiendo por que se pide :id cuando en la route se pasa el **:organization_id** y el **:gift_item_id** NO el ID del **gift_request** Para mi el test falla porque se esta llamando a destroy con un id que no se que es y no con el id del gift_request
Qué opinas?

### CODE REVIEW

File: app/models/wish_item.rb [hunk 1]
```diff
@@ -23,6 +26,10 @@ class WishItem < ActiveRecord::Base
 
   validate :check_finish_date, if: proc { finish_date.present? && self.finish_date_changed? }
 
+  def self.find_by_slug_or_id(param)
```
Comment (Dinuuu) [inline app/models/wish_item.rb:29]: why do you define this only for this class ?

Comment (vickymesalcorta) [inline app/models/wish_item.rb:29]: I also defined that method for Organization, User, and Gift Item. That method returns an instance given a "slug" or an id. (The slug is used for the pretty url, for example, "luciana-reznik"

Comment (Dinuuu) [inline app/models/wish_item.rb:29]: You are right, I didn't read it

**Comment by Dinuuu** [comment]
Si, si es asi como me decis lo de los tests, deberia fallar, pero no entiendo como andaba antes entonces

**Comment by vickymesalcorta** [comment]
@Dinuuu ahi lo corregí avisa asi mergeo

File: app/controllers/user_gift_requests_controller.rb [hunk 2]
```diff
@@ -5,7 +5,7 @@ class UserGiftRequestsController < ApplicationController
   before_action :validate_user, only: [:pending]
 
   def cancel
-    gift_request.destroy
+    @gift_request.destroy
```
Comment (vickymesalcorta) [inline app/controllers/user_gift_requests_controller.rb:8]: @Dinuuu ahi lo corregí y ademas le puse @gift_request porque asi lo nombra el método check_ownernship:

  def check_ownership
    request_to_cancel = gift_request_by_organization
    @gift_request = request_to_cancel || gift_request
    return render status: :forbidden,
                  text: "You don't own the item" unless gift_request.present? &&
                                                        gift_request.user_id == current_user.id
  end

**Comment by Dinuuu** [comment]
:+1:

**Pull request merged**